src/main/java/mate/academy/intro/service/user/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.intro.service.user.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.intro.dto.user.UserRegistrationRequest;
+import mate.academy.intro.dto.user.UserResponseDto;
+import mate.academy.intro.exception.RegistrationException;
+import mate.academy.intro.mapper.UserMapper;
+import mate.academy.intro.model.Role;
+import mate.academy.intro.model.Role.RoleName;
+import mate.academy.intro.model.User;
+import mate.academy.intro.repository.user.UserRepository;
+import mate.academy.intro.service.role.RoleService;
+import mate.academy.intro.service.user.UserService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+    private final RoleService roleService;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequest request) throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable to complete registration");
+        }
+        if (!request.getPassword().equals(request.getRepeatPassword())) {
+            throw new RegistrationException("Wrong password or repeatPassword...");
+        }
```

Comment:
You already have checked that password and repeatPassword are the same using @FieldMatch. 

```suggestion
```